### ISSUE TITLE
ping utility attempts to do blocking look up for the reverse DNS for EVERY ICMP packet if it fails

### ISSUE THREAD

**Issue by ryannathans** [iputils/iputils#414]
I ran `ping -6 aussiebroadband.com.au -i 0.2` and noticed that the output was definitely not every 200ms, it was more like 1 or 2 every few seconds, with random bursts.

So I fired up wireshark and (filtered to DNS only)
![image](https://user-images.githubusercontent.com/1307375/178892333-76205cb0-6fc9-4e49-b41b-49d8e4cf346a.png)

It looks like name resolution is failing and for every single ICMP packet there's a blocking request to get the hostname for the IP again.

This behavior does not occur with the `-n` flag, as expected, so that's a reasonable work around.

I don't think the intended behavior is to look up the hostname in a blocking manner for every single ping if it's failing.

**Comment by pevik** [comment]
@ryannathans Does it still happen? If yes, it should happen also on IPv4, right?

I was not able to reproduce it - running it longer time, but only single DNS PTR request:
```
$ ping -i 0.2 -6 aussiebroadband.com.au
PING aussiebroadband.com.au (2606:4700:10::6814:57f7) 56 data bytes
64 bytes from 2606:4700:10::6814:57f7: icmp_seq=1 ttl=54 time=34.1 ms
64 bytes from 2606:4700:10::6814:57f7: icmp_seq=2 ttl=54 time=22.8 ms
64 bytes from 2606:4700:10::6814:57f7: icmp_seq=3 ttl=54 time=27.7 ms
64 bytes from 2606:4700:10::6814:57f7: icmp_seq=4 ttl=54 time=25.5 ms
...
```

```
00:27:35.657116 IP x230.33112 > dns.google.domain: 41879+ A? aussiebroadband.com.au. (40)
00:27:35.657136 IP x230.33112 > dns.google.domain: 46571+ AAAA? aussiebroadband.com.au. (40)
00:27:35.758491 IP x230.50372 > dns.google.domain: 53272+ PTR? 7.f.7.5.4.1.8.6.0.0.0.0.0.0.0.0.0.0.0.0.0.1.0.0.0.0.7.4.6.0.6.2.ip6.arpa. (90)
```
```
ping -i 0.2  aussiebroadband.com.au 
PING aussiebroadband.com.au (172.67.18.20) 56(84) bytes of data.
64 bytes from 172.67.18.20: icmp_seq=1 ttl=53 time=24.2 ms
64 bytes from 172.67.18.20: icmp_seq=2 ttl=53 time=28.4 ms
64 bytes from 172.67.18.20: icmp_seq=3 ttl=53 time=22.3 ms
64 bytes from 172.67.18.20: icmp_seq=4 ttl=53 time=27.1 ms
64 bytes from 172.67.18.20: icmp_seq=5 ttl=53 time=22.5 ms
...
```
```
00:30:08.933808 IP x230.52552 > dns.google.domain: 62828+ A? aussiebroadband.com.au. (40)
00:30:08.933848 IP x230.52552 > dns.google.domain: 45672+ AAAA? aussiebroadband.com.au. (40)
00:30:09.018872 IP x230.52001 > dns.google.domain: 23276+ PTR? 247.87.20.104.in-addr.arpa. (44)
```

What am I missing? Some DNS caching?

First I thought we would need this change
https://github.com/pevik/iputils/commit/9225817281f8af73829091ceba5abfa74c5c1300 (from https://github.com/pevik/iputils/commits/ping/disable-dns-lookup-in-loop-ipv6/), but it's not needed.

```diff
diff --git ping/ping.c ping/ping.c
index b84ab7f..f6e0f15 100644
--- ping/ping.c
+++ ping/ping.c
@@ -1654,7 +1654,7 @@ int ping4_parse_reply(struct ping_rts *rts, struct socket_st *sock,
                        wrong_source = 1;
                if (gather_statistics(rts, (uint8_t *)icp, sizeof(*icp), cc,
                                      ntohs(icp->un.echo.sequence),
-                                     reply_ttl, 0, tv, pr_addr(rts, from, sizeof *from),
+                                     reply_ttl, 0, tv, pr_raw_addr(rts, from, sizeof *from),
                                      pr_echo_reply, rts->multicast, wrong_source)) {
                        fflush(stdout);
                        return 0;
diff --git ping/ping6_common.c ping/ping6_common.c
index eb3b0d2..a174195 100644
--- ping/ping6_common.c
+++ ping/ping6_common.c
@@ -866,7 +866,7 @@ int ping6_parse_reply(struct ping_rts *rts, socket_st *sock,
 
                if (gather_statistics(rts, (uint8_t *)icmph, sizeof(*icmph), cc,
                                      ntohs(icmph->icmp6_seq),
-                                     hops, 0, tv, pr_addr(rts, from, sizeof *from),
+                                     hops, 0, tv, pr_raw_addr(rts, from, sizeof *from),
                                      pr_echo_reply,
                                      rts->multicast, wrong_source)) {
                        fflush(stdout);
@@ -879,7 +879,7 @@ int ping6_parse_reply(struct ping_rts *rts, socket_st *sock,
                        return 1;
                if (gather_statistics(rts, (uint8_t *)icmph, sizeof(*icmph), cc,
                                      seq,
-                                     hops, 0, tv, pr_addr(rts, from, sizeof *from),
+                                     hops, 0, tv, pr_raw_addr(rts, from, sizeof *from),
                                      pr_niquery_reply,
                                      rts->multicast, 0))
                        return 0;

**Comment by ryannathans** [comment]
@pevik Seems like my distribution hasn't updated the utilities since my initial bug report, where it was the latest release at the time. Apologies for not posting the version in my initial report, not sure how I missed that.

```
ping -V                    
ping from iputils 20211215
```

I have confirmed the same issue existed for ipv4 too, with `ping -i 0.2 ryannathans.net` and looking at wireshark as it runs.

I downloaded and compiled the latest release, 20240117 and **the issue is not present anymore**. Perhaps it was fixed as a side effect of another patch? 

I noticed the output of the command has changed, perhaps that's related?

20211215:
`64 bytes from 2606:4700:10::6814:57f7 (2606:4700:10::6814:57f7): icmp_seq=25 ttl=60 time=2.87 ms`

20240117:
`64 bytes from 2606:4700:10::6814:57f7: icmp_seq=11 ttl=60 time=2.97 ms`
